nucypher/cli/commands/multisig.py

```diff
@@ -0,0 +1,357 @@
+class BlockchainOptions:
+
+    __option_name__ = 'blockchain_options'
+
+    def __init__(self, provider_uri, geth, poa, registry_filepath, light, network):
+        self.provider_uri = provider_uri
+        self.geth = geth
+        self.poa = poa
+        self.registry_filepath = registry_filepath
+        self.light = light
+        self.network = network
+
+    def get_registry(self, connect_blockchain: bool = False, emitter=None, debug=None):
+        if connect_blockchain:
+            self.connect_blockchain(emitter, debug)
+        if self.registry_filepath:
+            registry = LocalContractRegistry(filepath=self.registry_filepath)
+        else:
+            registry = InMemoryContractRegistry.from_latest_publication(network=self.network)
+        return registry
+
+    def connect_blockchain(self, emitter, debug):
+        try:
+            eth_node = None
+            if self.geth:
+                eth_node = get_provider_process()
+
+            # Note: For test compatibility.
+            if not BlockchainInterfaceFactory.is_interface_initialized(provider_uri=self.provider_uri):
+                BlockchainInterfaceFactory.initialize_interface(provider_uri=self.provider_uri,
+                                                                provider_process=eth_node,
+                                                                poa=self.poa,
+                                                                light=self.light,
+                                                                sync=False,
+                                                                emitter=emitter)
+
+            blockchain = BlockchainInterfaceFactory.get_interface(provider_uri=self.provider_uri)
+
+            emitter.echo(message="Reading Latest Chaindata...")
+            blockchain.connect()
+            return blockchain
+        except Exception as e:
+            if debug:
+                raise
+            click.secho(str(e), bold=True, fg='red')
+            raise click.Abort
+
+
+group_blockchain_options = group_options(
+    BlockchainOptions,
+    provider_uri=option_provider_uri(),
+    geth=option_geth,
+    poa=option_poa,
+    light=option_light,
+    registry_filepath=option_registry_filepath,
+    network=option_network,
+)
+
+
+class MultiSigOptions:
+    __option_name__ = 'multisig_options'
+
+    def __init__(self, checksum_address, signer_uri, hw_wallet):
+        self.checksum_address = checksum_address
+        self.signer_uri = signer_uri
+        self.hw_wallet = hw_wallet
+
+    def __create_executive(self, registry, transacting: bool = False) -> Executive:
+        client_password = None
+        if transacting and not self.hw_wallet:
+            client_password = get_client_password(checksum_address=self.checksum_address)
+        executive = Executive(checksum_address=self.checksum_address,
+                              registry=registry,
+                              signer=ClefSigner(self.signer_uri),
```

Comment:
Maybe not this PR - but if we eventually intend to have different types of signers, we may need to instead pass the object in as a parameter, as opposed to assuming a `ClefSigner`.